### ISSUE TITLE
Ratio and Proportion Description ready for implementation

### ISSUE THREAD

**Issue by Matthew-Moore240** [phetsims/ratio-and-proportion#87]
Lots of updates to the description are ready to be added. Ill make a check list of all the headings that need to be implemented, everything in red just notates that its been updated and is ready to go.

- [x] **H1: Explore Screen, Ratio and Proportion**

- [x] Static Screen Summary

- [x] Dynamic state-of-sim (qualitative):

- [x] Dynamic state-of-sim (quantitative):

- [x] Static sim-specific interaction hint in a paragraph:

- [x] **H3: Move Hands**

- [x] Left Hand, slider/Right Hand, slider 

- [x] Help Text For individual hands

- [x] Both Hands, moveable,

- [x] Help text for the both hands interaction 

- [x] **H3: Ratio Challenges** 

- [x] Challenge COMBOBOX

- [x] Help Text for ComboBox

- [x] **H3: Grid Options**

- [x] Grid Radio Button group

- [x] Grid Radio Button Help Text

**Challenge Maker Screen**

- [x] **H1: Challenge Maker Screen, Ratio and Proportion**

- [ ] Static Screen Summary

- [x] Dynamic state-of-sim Screen 2(quantitative):

- [x] Dynamic state-of-sim 2 (qualitative)

- [x] **H3: Grid Options**

- [x] Grid Radio Button group

- [x] Grid Radio Button Help Text

- [x] Grid Range ComboBox

- [x] Grid Range ComboBox Help Text

- [x] **H3: My Challenge**

- [x] Accordion Box

- [x] Help text for Accordion Box 

- [x] Number Spinners For Left and Right Hand

**Comment by zepumph** [comment]
Thanks! Working on it now.

https://docs.google.com/document/d/1HJvvGWYaUv9tlat9S7omQQVng8fGB4FuUhTEM1qx1z8/edit#

**Comment by zepumph** [comment]
I have implemented the basic stucture, labels, and help text for everything except the AccordionBox as a component (I did the content). 

* Next I will look into the screen summaries.

* Later I will review the response descriptions to see what may need implementing. For example, the last time I tried to move a hand with the keyboard, I didn't get description (perhaps from the composite listener), but I do get description from mouse input.

**Comment by zepumph** [comment]
All that seems to be left for static PDOM content (maybe this is called state description now?) are the screen summaries. I'll work on those updates now.

**Comment by zepumph** [comment]
I moved the common screen summary to the explore screen, and redid it so that it matched the design document. It wasn't totally trivial to add the dynamic value of what challenge we were on, so I left that constant with a TODO. Next is to implement that, and to add the screen summary for the create screen.

**Comment by zepumph** [comment]
- [ ] Grid range combo box needs the "Range" span next to the button, see Molarity.

**Comment by zepumph** [comment]
Still to do for screen summaries:
- [ ] (explore) dynamic state of `{{challenge 1}}` combo box
- [ ] (create) dynamic state of grid range combo box
- [ ] (create) support left/right values toggling between qualitative and quantitative.

**Comment by zepumph** [comment]
>Grid range combo box needs the "Range" span next to the button, see Molarity.

I originally had this, but removed it during this issue because the design doc didn't include the "Range:" piece. I've added it back in now, and it matches the Molarity implementation.

**Comment by zepumph** [comment]
>(create) support left/right values toggling between qualitative and quantitative.

It is not yet clear what the quantitative value for these should be. I asked for clarification in the design doc, but for now here is a patch that is on its way, but would need some gridDescriber work to return a string for use by the bullet

<details>

```diff

Index: js/common/view/RatioAndProportionScreenView.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/common/view/RatioAndProportionScreenView.js	(revision 7f2caf19baf1da411c01d10c0e05aba575daceeb)
+++ js/common/view/RatioAndProportionScreenView.js	(date 1594931056402)
@@ -74,7 +74,7 @@
     const gridDescriber = new GridDescriber( model.valueRange, options.gridBaseUnitProperty );
 
     // @protected (read-only)
-    this.ratioDescriber = new RatioDescriber( model, gridDescriber );
+    this.ratioDescriber = new RatioDescriber( model, gridViewProperty, gridDescriber );
 
     // @protected
     this.gridViewProperty = gridViewProperty;
Index: js/common/view/RatioDescriber.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- js/common/view/RatioDescriber.js	(revision 7f2caf19baf1da411c01d10c0e05aba575daceeb)
+++ js/common/view/RatioDescriber.js	(date 1594931056397)
@@ -7,6 +7,7 @@
 import StringUtils from '../../../../phetcommon/js/util/StringUtils.js';
 import ratioAndProportion from '../../ratioAndProportion.js';
 import ratioAndProportionStrings from '../../ratioAndProportionStrings.js';
+import GridView from './GridView.js';
 
 const ratioDescriptionPatternString = ratioAndProportionStrings.a11y.ratio.descriptionPattern;
 
@@ -44,9 +45,10 @@
 
   /**
    * @param {RatioAndProportionModel} model
+   * @param {Property.<GridView>} gridViewProperty
    * @param {GridDescriber} gridDescriber
    */
-  constructor( model, gridDescriber ) {
+  constructor( model, gridViewProperty, gridDescriber ) {
 
     // @private - from model
     this.ratioFitnessProperty = model.ratioFitnessProperty;
@@ -56,10 +58,42 @@
     this.ratioFitnessProperty = model.ratioFitnessProperty;
     this.fitnessRange = model.fitnessRange;
 
+    // @private
+    this.gridViewProperty = gridViewProperty;
+
     // @private
     this.gridDescriber = gridDescriber;
   }
 
+
+  getLeftPosition( gridView ) {
+  }
+
+  getRightPosition() {
+
+  }
+
+  getHandPosition( valueProperty, gridView ) {
+    return GridView.describeQualitative( gridView ) ? this.getQualitativePointerPosition( valueProperty ) :
+           this.gridDescriber.getRelativePositionAndGridNumberForProperty( valueProperty );
+  }
+
+  /**
+   * @public
+   * @returns {{grid: number, relativePosition: string}}
+   */
+  getLeftQuantitativePositionObject() {
+    return this.gridDescriber.getRelativePositionAndGridNumberForProperty( this.leftValueProperty );
+  }
+
+  /**
+   * @public
+   * @returns {{grid: number, relativePosition: string}}
+   */
+  getRightQuantitativePositionObject() {
+    return this.gridDescriber.getRelativePositionAndGridNumberForProperty( this.rightValueProperty );
+  }
+
   /**
    * @public
    * @returns {string}
@@ -113,22 +147,6 @@
     }
   }
 
-  /**
-   * @public
-   * @returns {{grid: number, relativePosition: string}}
-   */
-  getLeftQuantitativePositionObject() {
-    return this.gridDescriber.getRelativePositionAndGridNumberForProperty( this.leftValueProperty );
-  }
-
-  /**
-   * @public
-   * @returns {{grid: number, relativePosition: string}}
-   */
-  getRightQuantitativePositionObject() {
-    return this.gridDescriber.getRelativePositionAndGridNumberForProperty( this.rightValueProperty );
-  }
-
   /**
    * Get an index for the current fitness of the ratio, relative to the number of possible fitness descriptions. Low
    * values are very far from the ratios and high values are closer to the ratio from 0 to RATIO_FITNESS_STRINGS_CAPITALIZED.length.

**Comment by zepumph** [comment]
>Static sim-specific interaction hint in a paragraph:

I'm pretty sure this is an element of the screen summary that is just included from common code, like: 
>If needed, check out keyboard shortcuts under Sim Resources.

If so, then nothing needs to be done for this.

**Comment by zepumph** [comment]
All that I have left for this issue if for @Matthew-Moore240 et all to respond to comments from the google doc. The biggest one is what the quantitative screen summary bullets should look like for the left and right hands.

**Comment by zepumph** [comment]
I just spoke with @Matthew-Moore240, and he showed me a table in the design doc that I wasn't looking at. It is exactly the quantitative description that I was looking for. I will add that, and then ping him to review this implementation.

**Comment by zepumph** [comment]
After the above commits, all dynamic pieces of the screen summary should now be working as expected. @Matthew-Moore240 please review, and please create a new issue to discuss any changes or additions needed for context and object responses (i.e. alerts and aria-valuetext).

**Comment by emily-phet** [comment]
@zepumph and @Matthew-Moore240 
Is the current description up-to-date? I see some things that I think our older in the current a11y view on master (e.g., the scene summary for "Explore" describes 5 challenges instead of three).

If so @Matthew-Moore240 do you need any help going through and looking for issues? I have some time today if you'd like to meet and work on that together, in preparation for making some GitHub issues and for discussing tomorrow,

**Comment by zepumph** [comment]
- [ ] Hide the left {{number}}, right {number} in the create screen when the accordion box is hidden (linking the visual implementation with the pdom)

**Comment by zepumph** [comment]
@Matthew-Moore240 and I went through comments I made in the design doc, and made the above change to do. Implemented in https://github.com/phetsims/ratio-and-proportion/commit/7f267f245ee12c1292b648ca2ca85812943cb00f. Unassigning. 

I think all that is left for the PDOM is to get the second screen's summary in place. Thanks @Matthew-Moore240!

**Comment by Matthew-Moore240** [comment]
Made new issues for a couple of things @emily-phet mentioned. Everything else looks great any new updates will be put in there own issues. Amazing work @zepumph. Closing
